include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    Boole MEZZ_LIB IsAlphanumeric(const Char8 ToCheck);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // String Manipulation
+
+    /// @brief Converts all lower case characters in a character range to upper case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToUpperCase(StrIter Begin, StrIter End);
+    /// @brief Create a copy of the String that is upper case.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in all upper case.
+    String MEZZ_LIB UpperCaseCopy(String Source);
+    /// @brief Converts all upper case characters in a character range to lower case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToLowerCase(StrIter Begin, StrIter End);
+    /// @brief Create a copy of the String that is lower case.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in all lower case.
+    String MEZZ_LIB LowerCaseCopy(String Source);
+    /// @brief Converts the first letter of each word to upper case and all other letters to lower case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToCamelCase(StrIter Begin, StrIter End);
+    /// @brief Gets a copy of a String with the first letter of each word upper cased and other letters lower cased.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in Camel Case.
+    String MEZZ_LIB CamelCaseCopy(String Source);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Size Modifying String Manipulations
+
+    /// @brief Trims all whitespaces and tabs from a one or both sides of a String.
+    /// @param Source The String to be modified.
+    /// @param Left Whether or not to trim the left side of the String.
+    /// @param Right Whether or not to trim the right side of the String.
+    void MEZZ_LIB Trim(String& Source, const Boole Left = true, const Boole Right = true);
+    /// @brief Trims all whitespaces and tabs from a one or both sides of a String.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @param Left Whether or not to trim the left side of the String.
+    /// @param Right Whether or not to trim the right side of the String.
+    String MEZZ_LIB TrimCopy(String Source, const Boole Left = true, const Boole Right = true);
```

Comment:
I am used to seeing this as 3 different functions instead of parameters on one. LeftTrim, RightTrim and Trim.

Also a version of this could operate on iterator ranges like std::erase and return an iterator to the new end of the range or the end of the erased range,.

Reply:
I have modified it to have 3 versions.  The Non-copy version operates like std::erase as you suggested.